src/node/node_state.h

```diff
@@ -1513,27 +1513,55 @@ namespace ccf
 
     SessionMetrics get_session_metrics() override
     {
-      SessionMetrics sm;
-      rpcsessions->get_stats(sm.active, sm.peak, sm.soft_cap, sm.hard_cap);
-      return sm;
+      return rpcsessions->get_session_metrics();
     }
 
   private:
-    crypto::SubjectAltName get_subject_alt_name()
-    {
-      // If a domain is passed at node creation, record domain in SAN for node
-      // hostname authentication over TLS. Otherwise, record IP in SAN.
-      bool san_is_ip = config.domain.empty();
-      return {san_is_ip ? config.node_info_network.rpchost : config.domain,
-              san_is_ip};
+    bool is_ip(const std::string& hostname)
+    {
+      // IP address components are purely numeric. DNS names may be largely
+      // numeric, but at least the final component (TLD) must not be
+      // all-numeric. So this distinguishes "1.2.3.4" (and IP address) from
+      // "1.2.3.c4m" (a DNS name). "1.2.3." is invalid for either, and will
+      // throw. Attempts to handle IPv6 by also splitting on ':', but this is
+      // untested.
+      const auto final_component =
+        nonstd::split(nonstd::split(hostname, ".").back(), ":").back();
+      if (final_component.empty())
+      {
+        throw std::runtime_error(fmt::format(
+          "{} has a trailing period, is not a valid hostname",
+          final_component));
+      }
+      for (const auto c : final_component)
+      {
+        if (c < '0' || c > '9')
+        {
+          return false;
+        }
+      }
+
+      return true;
     }
 
     std::vector<crypto::SubjectAltName> get_subject_alternative_names()
```

Comment:
This used to be "everything passed to `--san`, plus one extra element (either the IP address on `--rpc-address`, or the DNS name on `--domain`". It is now "everything passed to `--san`, unless `--san` is empty in which case the default is each RPC address from `--rpc-interface` (auto-detected as IP address or DNS name)". I think that gives operators everything they need (the default should work for local testing, then you override it with `--san` for deployment, but needs documenting.